mp/src/game/client/momentum/ui/HUD/hud_speedometer.cpp

```diff
@@ -262,159 +257,62 @@ void CHudSpeedMeter::OnThink()
 
         switch (mom_hud_speedometer_units.GetInt())
         {
-        case 2:
-            // 1 unit = 19.05mm -> 0.01905m -> 0.00001905Km(/s) -> 0.06858Km(/h)
-            vel *= 0.06858f;
-            hvel *= 0.06858f;
-            lastJumpVel *= 0.06858f;
+        case 2: // KM/H
+            vel *= m_flKPHfactor; hvel *= m_flKPHfactor; lastJumpVel *= m_flKPHfactor;
             m_pUnitsLabel->SetText(L"KM/H");
             break;
-        case 3:
-            // 1 unit = 0.75", 1 mile = 63360. 0.75 / 63360 ~~> 0.00001184"(/s) ~~> 0.04262MPH
-            vel *= 0.04262f;
-            hvel *= 0.04262f;
-            lastJumpVel *= 0.04262f;
+        case 3: // MPH
+            vel *= m_flMPHfactor; hvel *= m_flMPHfactor; lastJumpVel *= m_flMPHfactor;
             m_pUnitsLabel->SetText(L"MPH");
             break;
-        case 4:
+        case 4: // Normalized units of energy
         {
-            // Normalized units of energy
             const auto gravity = sv_gravity.GetFloat();
-            vel = (pPlayer->GetAbsVelocity().LengthSqr() / 2.0f +
-                   gravity * (pPlayer->GetLocalOrigin().z - m_pRunEntData->m_flLastJumpZPos)) / gravity;
+            vel = (velocity.LengthSqr() / 2.0f + gravity * (pPlayer->GetLocalOrigin().z - m_pRunEntData->m_flLastJumpZPos)) / gravity;
             hvel = vel;
             m_pUnitsLabel->SetText(L"Energy");
             break;
         }
-        case 1:
+        case 1: // UPS
```

Comment:
Could probably go ahead and make a mom_shareddefs enum with the units types as the name:

```
enum SpeedometerUnitsType_t
{
    SPEEDO_UNITS_TYPE_UPS = 1, // Units per second
    SPEEDO_UNITS_TYPE_MPH,     // Miles per hour
 . . . etc
}
```